e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js

```diff
@@ -0,0 +1,122 @@
+import relative from '../../relativeURL';
+
+context(`${PDC_TEAMS.PDC_RECONCILE} users can remove fees from payments`, () => {
+  const reportId = 12;
+  const paymentId = 15;
+  const paymentCurrency = CURRENCY.GBP;
+  const paymentAmount = 100;
+
+  const paymentDateDay = '1';
+  const paymentDateMonth = '1';
+  const paymentDateYear = '2024';
+  const datePaymentReceived = new Date(`${paymentDateYear}-${paymentDateMonth}-${paymentDateDay}`);
+  const paymentReference = 'A payment reference';
+
+  const feeRecordIds = [1, 2];
+
+  const utilisationReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS)
+    .withId(reportId)
+    .withBankId('961')
+    .withDateUploaded(new Date())
+    .build();
+
+  const aPaymentWithAmount = (amount) =>
+    PaymentEntityMockBuilder.forCurrency(paymentCurrency)
+      .withId(paymentId)
+      .withAmount(amount)
+      .withDateReceived(datePaymentReceived)
+      .withReference(paymentReference)
+      .build();
+
+  const aFeeRecordWithAmountStatusAndPayments = (id, amount, status, payments) =>
+    FeeRecordEntityMockBuilder.forReport(utilisationReport)
+      .withId(id)
+      .withStatus(status)
+      .withFeesPaidToUkefForThePeriod(amount)
+      .withFeesPaidToUkefForThePeriodCurrency(paymentCurrency)
+      .withPaymentCurrency(paymentCurrency)
+      .withPayments(payments)
+      .build();
+
+  beforeEach(() => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [utilisationReport]);
+
+    const payment = aPaymentWithAmount(paymentAmount);
+
+    const feeRecords = feeRecordIds.map((id) => aFeeRecordWithAmountStatusAndPayments(id, paymentAmount, FEE_RECORD_STATUS.MATCH, [payment]));
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, feeRecords);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`/utilisation-reports/${reportId}/edit-payment/${paymentId}`);
+  });
+
+  it('should display errors when form submitted with invalid fee selections and persist the selected fees', () => {
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+    pages.utilisationReportEditPaymentPage.errorSummary().contains('Select fee or fees to remove from the payment');
+
+    feeRecordIds.forEach((id) => {
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).check();
+    });
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+    pages.utilisationReportEditPaymentPage.errorSummary().contains('You cannot remove all the fees. Delete the payment instead.');
+    feeRecordIds.forEach((id) => {
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).should('be.checked');
+    });
+  });
+
+  it('should remove the selected fees from the payment after clicking the remove selected fees button', () => {
```

Comment:
Could we possibly check that the other fee records still exist after we have removed this specific fee?

Reply:
Good point - made the change in 1018f9926faf81d4ac74ff707971a1e384418564 thanks